aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,119 @@
 
-## Services Available in Startup
+The web host provides some services that are available to the `Startup` class constructor. The app adds additional services via `ConfigureServices`. Both the host and app services are then available in `Configure` and throughout the application.
 
-ASP.NET Core dependency injection provides services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
+## The Configure method
 
-Below are some of the services typically requested by `Startup` methods:
+The [Configure](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configure) method is used to specify how the app responds to HTTP requests. The request pipeline is configured by adding [middleware](xref:fundamentals/middleware) components to an [IApplicationBuilder](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder) instance that's provided by [dependency injection](xref:fundamentals/dependency-injection). `IApplicationBuilder` is available to the `Configure` method, but it isn't registered in the service container. Hosting creates an `IApplicationBuilder` and passes it directly to `Configure` ([reference source](https://github.com/aspnet/Hosting/blob/release/2.0.0/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L179-L192)).
 
-* In the constructor:  `IHostingEnvironment`, `ILogger<Startup>`
-* In `ConfigureServices`:  `IServiceCollection`
-* In `Configure`:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
+The [ASP.NET Core templates](/dotnet/core/tools/dotnet-new) configure the pipeline with support for a developer exception page, [BrowserLink](http://vswebessentials.com/features/browserlink), error pages, static files, and ASP.NET MVC:
 
-Any services added by the ``WebHostBuilder`` ``ConfigureServices`` method may be requested by the ``Startup`` class constructor or its ``Configure`` method. Use `WebHostBuilder` to provide any services you need during `Startup` methods.
+[!code-csharp[Main](../common/samples/WebApplication1DotNetCore2.0App/Startup.cs?range=28-48&highlight=5,6,10,13,15)]
 
-## The Configure method
+Each `Use` extension method adds a middleware component to the request pipeline. For instance, the `UseMvc` extension method adds the [routing middleware](xref:fundamentals/routing) to the request pipeline and configures [MVC](xref:mvc/overview) as the default handler.
+
+Additional services, such as `IHostingEnvironment` and `ILoggerFactory`, may also be specified in the method signature. When specified, additional services are injected if they're available.
+
+For more information on how to use `IApplicationBuilder`, see [Middleware](xref:fundamentals/middleware).
+
+## Convenience methods
```

Comment:
This is a good sample that could be used here or in Hosting. Hosting may be more appropriate as these are IWebHostBuilder extensions.

Reply:
@Rick-Anderson There's a blurb over there that links to this startup topic, but do you want to ...
1. Leave as is?
2. Move this over to Hosting?
3. Duplicate this over in Hosting?
4. Improve the linked content over in Hosting that points over here?

That text in Hosting currently goes ...

> When setting up a host, you can provide \[Configure](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderextensions.configure?view=aspnetcore-1.1) and \[ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilder.configureservices?view=aspnetcore-1.1) methods. If you specify a \`Startup\` class, it must define a \`Configure\` method. For more information, see \[Application Startup in ASP.NET Core](startup.md).

Reply:
>Improve the linked content over in Hosting that points over here?

Make sure it's well exposed.

Reply:
Ok ... I'll take care of this when I get back to the house. I'm the driver on the family Friday night shopping run. ... ☔️ 🌧 and it's 40 degrees and raining out there. This is going to suck.